lib/logger/lib/logger.ex

```diff
@@ -592,4 +592,30 @@ defmodule Logger do
 
   defp notify(:sync, msg),  do: GenEvent.sync_notify(Logger, msg)
   defp notify(:async, msg), do: GenEvent.notify(Logger, msg)
+
+  defp handle_unused_variable_warnings(data, caller) do
+    vars_in_scope = Enum.map(caller.vars, fn({name, _value}) -> name end)
```

Comment:
Don't map. Keep the whole `caller.vars`. Below, you do: `if {name, scope} in caller.vars`.